src/core/service.c

```diff
@@ -2695,6 +2698,13 @@ static void service_sigchld_event(Unit *u, pid_t pid, int code, int status) {
                                 /* Fall through */
 
                         case SERVICE_RUNNING:
+                                if (s->type == SERVICE_NOTIFY) {
```

Comment:
this looks incorrect. This case statement is after all reached when the service is currently in SERVICE_RUNNING state, and the main process exits. In that case we should definitely consider the service shutting down.

I have the suspicion you wanted to place this if block right before this case statement, i.e. where the `/* Fall through */ is, in the SERVICE_START block, no?